File: test/extensions/language/parameterized_types/cpp/BytesParamTest.cpp
#include "gtest/gtest.h"
#include "parameterized_types/bytes_param/BytesParam.h"
#include "zserio/SerializeUtil.h"

namespace parameterized_types
{
namespace bytes_param
{

using allocator_type = BytesParam::allocator_type;
template <typename T>
using vector_type = zserio::vector<T, allocator_type>;

TEST(BytesParamTest, writeRead)
{
    BytesParam bytesParam(vector_type<uint8_t>{{0xCA, 0xFE}}, Parameterized(13));

    auto bitBuffer = zserio::serialize(bytesParam);
    auto readBytesParam = zserio::deserialize<BytesParam>(bitBuffer);

    ASSERT_EQ(bytesParam, readBytesParam);
}

} // namespace bytes_param
} // namespace parameterized_types

File: test/extensions/language/parameterized_types/python/BytesParamTest.py
import zserio

import ParameterizedTypes


class BytesParamTest(ParameterizedTypes.TestCase):
    def testWriteRead(self):
        bytesData = bytearray([0xCA, 0xFE])
        bytesParam = self.api.BytesParam(bytesData, self.api.Parameterized(bytesData, 13))

        bitBuffer = zserio.serialize(bytesParam)
        readBytesParam = zserio.deserialize(self.api.BytesParam, bitBuffer)
        self.assertEqual(bytesParam, readBytesParam)
